### PR TITLE
`bowtie latest-report` command

### PR DESCRIPTION
**Pull request by adwait-godbole**
@Julian tests for this will rely on the http mocking part i suppose ?

----
📚 Documentation preview 📚: https://bowtie-json-schema--1221.org.readthedocs.build/en/1221/

### CODE REVIEW

File: bowtie/_cli.py [hunk 1]
```diff
@@ -1357,6 +1359,32 @@ async def filter_dialects(
             break
 
 
+@subcommand
+@click.option(
```
Comment (Julian) [inline bowtie/_cli.py:1363]: This is essentially redefining `@DIALECT`.

Comment (adwait-godbole) [reply bowtie/_cli.py:1363]: But the `@DIALECT` doesn’t have a default 2020-12 on it I suppose.

File: bowtie/_cli.py [hunk 2]
```diff
@@ -1357,6 +1359,32 @@ async def filter_dialects(
...
+    ),
+)
+def latest_report(dialect: Dialect):
+    """
+    Output latest Bowtie report for a given dialect.
+
+    If no dialect provided, it outputs Bowtie's latest
+    dialect's latest report.
+    """
+    for line in asyncio.run(dialect.latest_report()).iter_lines():
```
Comment (Julian) [inline bowtie/_cli.py:1384]: You don't really need lines here, you're just writing the contents unchanged -- possibly there's something even more straightforward that isn't occurring to me but at least the below makes fewer assumptions:

```python
def latest_report(...):
    async def write(response):
        async for chunk in (await response).aiter_bytes():
            click.echo(chunk)
    asyncio.run(write(dialect.latest_report()))

```

File: bowtie/_cli.py [hunk 3]
```diff
@@ -816,9 +819,11 @@ def statistics(
     format: _F,
 ):
     """
-    Show summary statistics for a previous report.
+    Show summary statistics for a Bowtie generated report.
 
-    If no report provided, defaults to Bowtie's latest dialect's latest report.
+    If no report provided, it shows statistics for Bowtie's latest dialect's
+    latest report. For getting statistics on other dialect reports, do
+    `bowtie latest-report --dialect {YOUR_DESIRED_DIALECT} | bowtie statistics`
```
Comment (adwait-godbole) [inline bowtie/_cli.py:826]: @Julian is this fine as a helper text 😅?

File: bowtie/_cli.py [hunk 4]
```diff
@@ -1364,6 +1369,23 @@ async def filter_dialects(
...
 
 
+@subcommand
+@DIALECT
+def latest_report(dialect: Dialect):
+    """
+    Get latest Bowtie generated report for a dialect.
+
+    If no dialect provided, it outputs Bowtie's latest
+    dialect's latest generated report.
```
Comment (adwait-godbole) [inline bowtie/_cli.py:1379]: Also this one.

**Comment by Julian** [comment]
Merged with some minor changes, thanks again -- though I see the statistics tests we added are broken, specifically they only pass when run from a machine in the UTC timezone (like the GitHub actions runners), but fail locally. The timezone handling is off, we probably should just use dateutil or pendulum which are the usual solutions there, but I'll have to look tomorrow.

**Pull request merged**